insights/core/__init__.py

```diff
@@ -1542,3 +1542,150 @@ def __init__(self, *args, **kwargs):
         deprecated(AttributeDict, "Please set attributes explicitly.")
         super(AttributeDict, self).__init__(*args, **kwargs)
         self.__dict__ = self
+
+
+class ModInfo(Parser):
+    """
+    This will parase the inforamtion about a kernel module, the module
+    info will be stored in dictionary format.
+
+    Sample output:
+        filename:       /lib/modules/3.10.0-993.el7.x86_64/kernel/drivers/net/ethernet/intel/i40e/i40e.ko.xz
+        version:        2.3.2-k
+        license:        GPL
+        description:    Intel(R) Ethernet Connection XL710 Network Driver
+        author:         Intel Corporation, <e1000-devel@lists.sourceforge.net>
+        retpoline:      Y
+        rhelversion:    7.7
+        srcversion:     DC5C250666ADD8603966656
+        alias:          pci:v00008086d0000158Bsv*sd*bc*sc*i*
+        alias:          pci:v00008086d0000158Asv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001588sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001587sv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037D3sv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037D2sv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037D1sv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037D0sv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037CFsv*sd*bc*sc*i*
+        alias:          pci:v00008086d000037CEsv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001589sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001586sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001585sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001584sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001583sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001581sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001580sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001574sv*sd*bc*sc*i*
+        alias:          pci:v00008086d00001572sv*sd*bc*sc*i*
```

Comment:
It's just a sample, It's not so necessary to paste the complete output to the docstring, please remove some duplicate lines.

Reply:
Done